File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left=0,right=height.size()-1,result=0;
        while(left<right)
        {
            result=max(result,min(height[left],height[right])*(right-left));
            if(height[left]>height[right])
            {
                right--;
            }
            else
            {
                left++;
            }
        }
        return result;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int low=0,high=height.length-1;
        int ans=0;
        while(low<high)
        {
            ans=Math.max(ans,Math.min(height[low],height[high])*(high-low));
            if(height[low]<height[high])
            {
                low++;
            }
            else
            {
                high--;
            }
        }
        return ans;
    }
}